### ISSUE TITLE
[WotLK] quest 4743 and quest 5122

### ISSUE THREAD

**Issue by apo126** [unified-db/Database#186]
item 12300 cannot be used to control the npc，quest 4743 cannot be completed.

quest 5122 cannot complete, because the item 12845 cannot be retrieved from the box.

**Comment by Grz3s** [comment]
Regarding to 1st quest:
I made a fix for it.. 90% of it works fine exept one thing:
possesion spell can be casted anytime on npc :/ this would be to easy :]
so:
Question to @xfurry 
Is it possibleto create a SD2 script for creature.. that spell "16053" from item "y" ...  can be casted only if creature 10321 is down to 10% HP?

**Comment by xfurry** [comment]
@Grz3s I think there is a way to check for health percent requirements on item user or spell cast.
I need to see exactly how it's this done for other spells / items.

**Comment by Grz3s** [comment]
I'll probably fix second quest soon* ... and then ... i'll open new issue with this bug on sd2 side .. if you dont mind.

**Comment by Grz3s** [comment]
@xfurry 
> @Grz3s can you please point me to some documentation to help me understand what item and what spell needs to be fixed. Thanks

well, im not sure ... if *fixed* is right word here:D... I wouldnt say smth is broken... it just missing "option" in DB or Acid... so thats why I asked you (sd2) ;)

item: 
http://www.wowhead.com/item=12300/orb-of-draconic-energy
spell from this item:
http://www.wowhead.com/spell=16053/dominion-of-soul
you can see that this spell has a target set in DB:
```sql
(16053,1,10321,0),
```

Player should see this textemote: '%s\'s will falters.' - when creature 10321 has 10%HP and from that moment, our item(spell) could be used to take control on him.
So basically We need sd2 script - to do this trick :D

**Comment by Grz3s** [comment]
second script leads to another problem ... 
guest giver .. offers us... quest that should be available only after killing "Baron Rivendare"
I need to find more info... how it was done on Blizz,,
Maybe @cala has some info?

**Comment by cala** [comment]
Chance are high that this needs to be handled by SD2 because it starts a script depending on the encounters status of the instance (Baron killed or not).
How it should be done:
- Any player speaks with Aurius (10917) and take repeatable quest 5122
- players go to crusade part of the instance and get item 12845 (Medaillon of Faith)
- players hand it over to Aurius to complete the quest
- if Baron is still alive, next quest (5125) is proposed by Aurius
- upon fighting Baron, Aurius comes in and takes part to the fight: he dies when Baron dies and then quest 5125 is complete

**Comment by Grz3s** [comment]
hi @cala ..thx for respond...
quest 5122 - has ben fixed above...
quest 5125 - cannot be accepted (its one of type quests that u only need to speak with npc to complete it)... 
so i just Wonder... if:
- after 5122-- he should do smth. like walk off and despawn(i could remove his quest flag here) .. so players wont be able to complete second quest... and then spawn again with boss fight.
- create sd2 script as u said... to check if baron is still alive... etc.. 

I need to find ... if he despawns from first possition...
Got any old avis with this instance?:)

**Comment by cala** [comment]
Aurius does not despawn from the chapell. A duplicate of him is spawned during the encounter with the baron if the medaillon was brought to him.

**Comment by xfurry** [comment]
@Grz3s I'm pretty sure I saw quests where the item or spell can't be used on a creature unless it has HP lower than x%. I don't know any examples right now, but I'm thinking that we could use a similar way to get around your SD2 requirement.

**Comment by Rushor** [comment]
example: http://www.wowhead.com/quest=11590/abduction

**Comment by cala** [comment]
There are two different quests set in two different zones, it becomes hard to follow the thread.

Anyway, regarding the Stratholme quest/event: I intend in a not to distant futur to enhance/add SD2 scripts for Stratholme the same way I did for Blackrock Depths last autumn. So, I can add this part to my list.

**Comment by Grz3s** [comment]
@xfurry still around?
I have found smth interesting... regarding to:
```quote
Question to @xfurry 
 Is it possible to create a SD2 script for creature.. that spell "16053" from item "y" ... can be casted only if creature 10321 is down to 10% HP?
```

Spell = 29364 Encapsulate Voidwalker
Can be used only if it's target has 20%HP or less.... checked it in game and works fine..
So i checked it ..and:

```code
ID - 29364 Encapsulate Voidwalker
Duration: ID (62)  75000, 0, 75000
Interrupt Flags: 0x00000000, AuraIF 0x00000000, ChannelIF 0x00000000
TargetAuraState = 2 (AURA_STATE_HEALTHLESS_20_PERCENT)   ----- here
Chance = 101, charges - 0
```
I just Wonder if this aura ... could be implemented to our spell... somehow using sd2 ??

**Comment by xfurry** [comment]
> I just Wonder if this aura ... could be implemented to our spell... somehow using sd2 ??

I'm not sure, but I don't think we can do this in a nice way.
Probably we need to script the item, as you originally proposed.

**Comment by cala** [comment]
Quest 5125 and the related event are fixed by SD2 in https://github.com/cmangos/mangos-wotlk/commit/47b3c2499381c09e7f2ca02cda3fda4025117645
No change required DB side unless the template MovementType for NPC_AURIUS is different from 0.

I'll look later at the second issue (Emberstrife) if @xfurry does not beat me to it. :)

**Comment by Grz3s** [comment]
:+1:  @cala for your NPC_AURIUS script -- very good job;)
So .... 
The last issue here is:
"spell 16053 can be casted only if Emberstrife is down to 10%HP"

I could use ACiD and give him immune for SpellSchoolMask = 2 (MASK_HOLY)... rhis is a source of our spell
and then unblock it... when he is 10% ... this would work... but question is...
How this damage players like Priests ??  ... are they using a lot of holy? as a offensive spells?

**Comment by cala** [comment]
IMHO, using immunity to holy spell school is a hack as it will damage priests as you said but also paladins that use holy in a more offensive matter than priests.

I've started scripting the item in SD2 but did not come to something working right now. I may need @xfurry's help on the matter (I already must discuss with him at least two others SD2 topics).

**Comment by cala** [comment]
Will be closed by the WotLK backport of https://github.com/cmangos/mangos-classic/commit/bd1e0da20f7ceba0f6b3ddc7afab7dfd0c65c0ec once I've backported it.

**Comment by cala** [comment]
Can be closed by https://github.com/cmangos/mangos-wotlk/commit/9e594c148bd5cf4ecfe624493bf0964a52b673e2
